core/src/main/java/hudson/ExtensionList.java

```diff
@@ -418,6 +418,24 @@ private void fireOnChangeListeners() {
     }
 
     /**
+     * Convenience method allowing lookup of the only instance of a given type.
+     * Equivalent to {@code ExtensionList.lookup(Class).get(Class)} if there is one instance,
+     * and throws an {@code IllegalStateException} otherwise.
+     *
+     * @param type The type to look up.
+     * @return the singleton instance of the given type in its list.
+     *
+     * @since TODO
+     */
+    public static @Nonnull <U> U lookupSingleton(Class<U> type) {
+        ExtensionList<U> all = lookup(type);
+        if (all.size() != 1) {
+            throw new IllegalStateException("Expected 1 instance of " + type.getName() + " but got " + all.size());
```

Comment:
Needs to be Javadoced. `throws` would be also useful.
In the case of this method, I agree that `RuntimeException` is a right approach.

Reply:
`Javadoced` :smile: Love English language.

Reply:
@oleg-nenashev It's mentioned in https://github.com/jenkinsci/jenkins/pull/3021/files#diff-e3f986a0854b9b724f4e12690cdeeac3R423 and I see no point in adding it to the signature. It's a runtime exception after all.

Reply:
Still a good practice; helpful for Javadoc viewers.

```java
/**
 * …
 * @throws IllegalStateException if there are no instances, or more than one
 */
```